File: assignment 10 - C# practice problems/35.cs
//Write a c# program to compute the union of two arrays. 

using System;
using System.Collections.Generic;
using System.Linq;

namespace MyNameSpace
{
    class Program35
    {
        public static void ArayUnionDriverFunction()
        {

            var numArray1 = new List<int>() { 1, 2, 2, 5, 5, 5, 5, 5, 5, 5, 8, 7, 9, 9, 5, 9, 6 };
            var numArray2 = new List<int>() { 1, 2, 3, 4, 5 };
            Console.WriteLine("arrays are 1: [{0}] 2: [{1}]", string.Join(",", numArray1), string.Join(",", numArray2));

            var resultArray = arrayUnion(numArray1, numArray2);
            Console.WriteLine("Array Union is: [{0}]", string.Join(",", resultArray));


        }

        static List<int> arrayUnion(List<int> arrayParam1, List<int> arrayParam2)
        {

            var array = arrayParam1.Concat(arrayParam2).ToList();

            var uniqueArray = new List<int>();
            foreach (var item in array)
            {

                var itemFound = uniqueArray.IndexOf(item);

                if (itemFound == -1)
                {
                    uniqueArray.Add(item);
                }

            }

            return uniqueArray;
        }


    }
}

File: assignment 3 - js practice problems/web-dev-starter/35.js
//Write a JavaScript program to compute the union of two arrays. 
function arrayUnion(arrayParam1, arrayParam2) {

    array = arrayParam1.concat(arrayParam2);

    let uniqueArray = [];
    array.forEach((item) => {

        let itemFound = uniqueArray.findIndex((uniqueItem) => {

            return uniqueItem === item;

        })

        if (itemFound == -1) uniqueArray.push(item);
    })

    return uniqueArray;
}

function arrayUnionDriverFunction() {
    let arrayElement1 = [1, 2, 2, 5, 5, 5, 5, 5, 5, 5, 8, 7, 9, 9, 5, 9, 6];
    let arrayElement2 = [1, 2, 2, 100, 6, 8];

    console.log("arrays are: \n1: " + arrayElement1 + "\n2: " + arrayElement2);
    console.log("array Union is: " + arrayUnion(arrayElement1, arrayElement2));
}

arrayUnionDriverFunction();